packages/subscribable/src/__typetests__/data-publisher-typetest.ts

```diff
@@ -0,0 +1,60 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+
+import { DataPublisher, getDataPublisherFromEventEmitter } from '../data-publisher';
+import { TypedEventEmitter } from '../event-emitter';
+
+type ChannelMap = {
+    fall: null;
+    jump: { height: number };
+    run: { velocity: number };
+};
+const publisher = null as unknown as DataPublisher<ChannelMap>;
+
+// [DESCRIBE] getDataPublisherFromEventEmitter
+{
+    // It materializes listener signatures based on the events of the emitter
+    {
+        const eventEmitter = null as unknown as TypedEventEmitter<{ foo: CustomEvent<'bar'> }>;
+        const publisher = getDataPublisherFromEventEmitter(eventEmitter);
+        publisher.on('foo', data => {
+            data satisfies 'bar';
+        });
+    }
```

Comment:
Could we add a typetest for non-custom events? I'm guessing, based on the type definition below, that `null` will be provided as the first argument.

Something like this:

```ts
const eventEmitter = null as unknown as TypedEventEmitter<{ foo: SomeEvent }>;
const publisher = getDataPublisherFromEventEmitter(eventEmitter);
publisher.on('foo', (...args) => {
    args satisfies readonly [null]; // Is this right?
});
```

Reply:
Done!